### PR TITLE
kafka: check api_version

### PR DESCRIPTION
**Pull request by gabriel-tincu**
When enabling ZSTD, check the kafka api version , since the compression
algorithm requires a cluster version of 2.1.0 or higher

### CODE REVIEW

**Comment by hackaugusto** [comment]
This is done by `kafka-python` already:

https://github.com/dpkp/kafka-python/blob/9feeb79140ed10e3a7f2036491fc07573740c231/kafka/producer/kafka.py#L395-L396

File: journalpump/senders/kafka.py [hunk 1]
```diff
@@ -37,18 +37,21 @@ def _init_kafka(self):
...
         while self.running:
             try:
+                api_version = self.config.get("kafka_api_version", "0.9")
+                parsed = tuple(map(int, api_version.split(".")))
                 if self.kafka_producer:
                     self.kafka_producer.close()
                 # make sure the python client supports it as well
-                if zstd and "zstd" in KafkaProducer._COMPRESSORS:  # pylint: disable=protected-access
+                # pylint: disable=protected-access
+                if zstd and "zstd" in KafkaProducer._COMPRESSORS and parsed >= (2, 1, 0):
```
Comment (ivanyu) [inline journalpump/senders/kafka.py:46]: Maybe, use `LooseVersion`?

**Review by ivanyu — CHANGES_REQUESTED** [review]

**Comment by fingon** [comment]
@ivanyu is this still worth doing or should we close it?

**Comment by ivanyu** [comment]
I think we don't need to care about that old Kafka versions.

**Pull request closed without merging**